mlrun/pipelines/kfp/v1_8/mixins.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import kfp
+
+import mlrun
+from mlrun.pipelines.common.helpers import project_annotation
+from mlrun.pipelines.utils import apply_kfp
+
+
+class KfpAdapterMixin:
+    def apply(self, modify):
+        """
+        Apply a modifier to the runtime which is used to change the runtimes k8s object's spec.
+        Modifiers can be either KFP modifiers or MLRun modifiers (which are compatible with KFP). All modifiers accept
+        a `kfp.dsl.ContainerOp` object, apply some changes on its spec and return it so modifiers can be chained
+        one after the other.
+
+        :param modify: a modifier runnable object
+        :return: the runtime (self) after the modifications
+        """
+
+        # Kubeflow pipeline have a hook to add the component to the DAG on ContainerOp init
+        # we remove the hook to suppress kubeflow op registration and return it after the apply()
+        old_op_handler = kfp.dsl._container_op._register_op_handler
+        kfp.dsl._container_op._register_op_handler = lambda x: self.metadata.name
+        cop = kfp.dsl.ContainerOp("name", "image")
+        kfp.dsl._container_op._register_op_handler = old_op_handler
+
+        return apply_kfp(modify, cop, self)
+
+
+class PipelineProviderMixin:
+    # TODO: this method is similar enough across KFP versions to not need a mixin
```

Comment:
you mean across mlrun kfp versions? (1.8/2.0?)

Reply:
yeah, but I still need to look into it a bit more to be sure. Not everything is compatible yet.